src/types.rs

```diff
@@ -303,34 +325,44 @@ impl Id {
       public_keys : (pub_sign_key, pub_asym_key),
       secret_keys : (sec_sign_key, sec_asym_key),
       validation_token : validation_token,
-      name : NameType::new(digest.0)
+      name : NameType::new(digest.0),
+      original_name : NameType::new(digest.0),
```

Comment:
So I did advise Prakash to keep the original name field for now, and we can take it out later?  Would that be okay for you Peter?